lib/core.ps1

```diff
@@ -42,6 +51,81 @@ function Get-UserAgent() {
     return "Scoop/1.0 (+http://scoop.sh/) PowerShell/$($PSVersionTable.PSVersion.Major).$($PSVersionTable.PSVersion.Minor) (Windows NT $([System.Environment]::OSVersion.Version.Major).$([System.Environment]::OSVersion.Version.Minor); $(if($env:PROCESSOR_ARCHITECTURE -eq 'AMD64'){'Win64; x64; '})$(if($env:PROCESSOR_ARCHITEW6432 -eq 'AMD64'){'WOW64; '})$PSEdition)"
 }
 
+function load_cfg($file) {
+    if(!(Test-Path $file)) {
+        return $null
+    }
+
+    try {
+        return (Get-Content $file -Raw | ConvertFrom-Json -ErrorAction Stop)
+    } catch {
+        Write-Host "ERROR loading $file`: $($_.exception.message)"
+    }
+}
+
+$scoopConfig = load_cfg $configFile
+
+function get_config($name, $default) {
+    if($null -eq $scoopConfig.$name -and $null -ne $default) {
+        return $default
+    }
+    return $scoopConfig.$name
+}
+
+function set_config($name, $value) {
+    if(!$scoopConfig) {
+        ensure (Split-Path -Path $configFile) | Out-Null
+        $scoopConfig = @{ $name = $value }
+    } else {
+        if($value -eq [bool]::TrueString -or $value -eq [bool]::FalseString) {
+            $value = [System.Convert]::ToBoolean($value)
+        }
+        if($scoopConfig.PSObject.Properties.Item($name)) {
```

Comment:
```suggestion
        if($null -ne $scoopConfig.$name) {
```

If `$scoopConfig` don't have `$name`, it is `$null`. It's needless to check the properties.

Reply:
No `$null` value will be in config, since it is removed in L90-L92 below, so check isnull is enough.